src/plugins/discourse/weights.test.js

```diff
@@ -10,11 +18,31 @@ function getMockUser(t: number): User {
   };
 }
 
+function getConfig(s: ?Object): DiscourseConfig {
+  return configParser.parseOrThrow({
+    serverUrl: "https://sc.sc",
+    weights: {
+      defaultTagWeight: 1,
+      defaultCategoryWeight: 1,
+      tagWeights: {},
+      categoryWeights: {},
+      ...s,
+    },
+  });
+}
+
+function getWeights(c: DiscourseConfig): WeightsConfig {
+  if (!c.weights) {
+    throw new Error("missing weights");
+  }
+  return c.weights;
+}
```

Comment:
as discussed elsewhere: make the WeightsConfig mandatory and we don't need this function at all.